recirq/otoc/parallel_xeb.py

```diff
@@ -0,0 +1,723 @@
+# Copyright 2020 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Functions for performing parallel cross entropy benchmarking."""
+
+from typing import Sequence, List, Set, Tuple, Dict, Union
+
+import cirq
+import numpy as np
+import pybobyqa
+from matplotlib import pyplot as plt
+
+from recirq.otoc.utils import (bits_to_probabilities, angles_to_fsim,
+                               pauli_error_fit, generic_fsim_gate)
+
+_rot_ops = [
+    cirq.X ** 0.5, cirq.PhasedXPowGate(phase_exponent=0.25, exponent=0.5),
+    cirq.Y ** 0.5, cirq.PhasedXPowGate(phase_exponent=0.75, exponent=0.5),
+    cirq.X ** -0.5, cirq.PhasedXPowGate(phase_exponent=-0.75, exponent=0.5),
+    cirq.Y ** -0.5, cirq.PhasedXPowGate(phase_exponent=-0.25, exponent=0.5),
+]
+_rot_mats = [cirq.unitary(r) for r in _rot_ops]
+_fsim_angle_labels = ['theta', 'delta_plus', 'delta_minus_off_diag',
+                      'delta_minus_diag', 'phi']
+
+
+def default_interaction_sequence(
```

Comment:
Not much in the current code. I think it would be good to retain this as it is convenient for breaking gates in 2D into different layers for parallel XEB.

Reply:
This exists as `cirq.experiments._default_interaction_sequence`.

Reply:
That's right (I actually copied the function here into Cirq...). This is now deleted from ReCirq.